codegen/pyadritem.txt

```diff
@@ -1,400 +1,402 @@
+
+from .adr_utils import in_ipynb, table_attr, type_maps
+import webbrowser
+
+try:
+    from IPython.display import IFrame
+except ImportError:
+    pass
+
+
+# Generate the items for the ADR database
+class Item:
+
+    """Provides for creating an object that represents an Ansys Dynamic Reporting item.
+
+    Create an instance of this class for each item in the database that you want to
+    interact with. When the object is created, no type is set. The type, determined the
+    first time that you set the ``item_*`` attribute, cannot be changed.
+
+    This code creates an instance with the object ``my_txt`` as a text item:
+
+    >>> my_txt = adr_service.create_item()
+    >>> my_txt.item_text = '<h1>The test</h1>This is a text item'
+
+
+    The type of the item created in the preceding code cannot be changed. However,
+    the attributes describing the object can be reset at any time. These hanges are
+    automatically propagated into the database. The attributes described in the
+    following "Parameters" section can be used to control the rendering of these objects.
+
+    .. note::
+       These attributes mirror the generic data item attributes described in
+       `Data Items`_ in the documentation for Ansys Dynamic Reporting.
+
+    .. _Data Items: https://nexusdemo.ensight.com/docs/html/Nexus.html?DataItems.html
+
+    Parameters
+    ----------
+    service : ansys.dynamicreporting.core.Service, optional
+        Ansys Dynamic Reporting object that provides the connection to the database
+        that the item is to interact with. The default is ``None``.
+    obj_name : str, optional
+        Name of the item object in the database. The default is ``default``.
+    source : str, optional
+        Name of the source for the item in the database. The default is ``"ADR"``.
+
+
+    Examples
+    --------
+    Initialize the ``Service`` class inside an Ansys Dynamic Reporting service and
+    create an object as a text item::
+
+        import ansys.dynamicreporting.core as adr
+        adr_service = adr.Service(ansys_installation = r'C:\\Program Files\\ANSYS Inc\\v232')
+        adr_service.connect(url='http://localhost:8010')
+        my_txt = adr_service.create_item()
+        my_txt.item_text = '<h1>The test</h1>This is a text item'
+
+    """
+
+    def __init__(self, service: 'ADR' = None, obj_name: Optional[str] = "default",
+                 source: Optional[str] = "ADR") -> None:
+        self.item = None
+        self.serverobj = service.serverobj
+        self._url = None
+        self.logger = service.logger
+        self.source = source
+        self.obj_name = str(obj_name)
+        self.type = None
+        self.item_text = ""
+        """Text (HTML and LaTeX formatting)"""
+        self.item_image = None
+        """Image object (Image and PNG binary files)"""
+        self.item_scene = None
+        """3D scene (AVZ, PLY, SCDOC, GLB, and STL files)"""
+        self.item_animation = None

```

Comment:
@viseshrp These are the two lines I changed.  I added self.item_animation and the associated doc string.  This worked previously because assigning 'item.item_animation = "filename.mp4"' created the member and the push mechanism found the data via attribute inspection.

Reply:
The entire change is adding these two lines.